### PR TITLE
Add properties table for ease of filtering events

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds a properties table to GcnEvents to simplify filtering on events.

### CODE REVIEW

File: skyportal/handlers/api/gcn.py [hunk 1]
```diff
@@ -598,6 +639,54 @@ async def get(self, dateobs=None):
                     tag_subquery, GcnEvent.dateobs != tag_subquery.c.dateobs
                 )
 
+            if properties_filter is not None:
+                for prop_filt in properties_filter:
+                    prop_split = prop_filt.split(":")
+                    if not (len(prop_split) == 1 or len(prop_split) == 3):
+                        raise ValueError(
+                            "Invalid propertiesFilter value -- propotation filter must have 1 or 3 values"
```
Comment (profjsb) [inline skyportal/handlers/api/gcn.py:647]: propotation?

File: skyportal/handlers/api/gcn.py [hunk 2]
```diff
@@ -598,6 +639,54 @@ async def get(self, dateobs=None):
                     tag_subquery, GcnEvent.dateobs != tag_subquery.c.dateobs
                 )
 
+            if properties_filter is not None:
+                for prop_filt in properties_filter:
+                    prop_split = prop_filt.split(":")
+                    if not (len(prop_split) == 1 or len(prop_split) == 3):
```
Comment (profjsb) [inline skyportal/handlers/api/gcn.py:645]: the API spec makes it seem like there must be 3 values after split. How can you get one?

File: skyportal/handlers/api/gcn.py [hunk 3]
```diff
@@ -598,6 +639,54 @@ async def get(self, dateobs=None):
...
+                        elif op == "le":
+                            comp_function = operator.le
+                        elif op == "eq":
+                            comp_function = operator.eq
+                        elif op == "ne":
+                            comp_function = operator.ne
+                        elif op == "ge":
+                            comp_function = operator.ge
+                        elif op == "gt":
+                            comp_function = operator.gt
```
Comment (profjsb) [inline skyportal/handlers/api/gcn.py:674]:
```suggestion
                        comp_function = getattr(operator, op)
```

File: skyportal/tests/api/test_gcn.py [hunk 4]
```diff
@@ -168,13 +185,27 @@ def test_gcn_from_moc(super_admin_token, view_only_token):
...
     data = data["data"]
     assert "2022-06-18T18:31:12" in [event["dateobs"] for event in data["events"]]
 
+    params = {'propertiesFilter': 'BNS: 0.5: gt'}
+    status, data = api('GET', 'gcn_event', token=super_admin_token, params=params)
+    assert status == 200
+    data = data["data"]
+    assert "2022-06-18T18:31:12" in [event["dateobs"] for event in data['events']]
+
+    params = {'propertiesFilter': 'BNS: 0.5: lt'}
```
Comment (profjsb) [inline skyportal/tests/api/test_gcn.py:203]: Test comma-separated filtering criteria?

File: skyportal/handlers/api/gcn.py [hunk 5]
```diff
@@ -598,6 +639,54 @@ async def get(self, dateobs=None):
                     tag_subquery, GcnEvent.dateobs != tag_subquery.c.dateobs
                 )
 
+            if properties_filter is not None:
```
Comment (profjsb) [inline skyportal/handlers/api/gcn.py:642]: I dont see how comma-separated criteria are parsed and used here

Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:642]: We do the split and the loop around line 545.

File: skyportal/tests/api/test_gcn.py (hunk 4 above)
Comment (mcoughlin) [reply skyportal/tests/api/test_gcn.py:203]: good idea.

File: skyportal/handlers/api/gcn.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:674]: ah nifty.

File: skyportal/handlers/api/gcn.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:647]: oops.

File: skyportal/handlers/api/gcn.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/gcn.py:645]: adding clarification to the spec.

**Comment by guynir42** [comment]
@mcoughlin Let me know if and when you want me to review this.

**Comment by mcoughlin** [comment]
@guynir42 should be ready for a look

**Review by guynir42 — COMMENTED** [review]
I wonder what is the advantage of having this as a separate table from the GCNs themselves. I see the similarities to PhotStats but this seems simple enough to just put the properties on the object itself. 
Unless there's some use cases where you can have multiple properties connected to each GCN?

**Comment by mcoughlin** [comment]
@guynir42 Yeah exactly, they update in time.

**Comment by guynir42** [comment]
@mcoughlin Then why not just override the existing properties? If you are always going to look at the most recent? Or is this useful because we want to keep a history of the changes like in redshift?

**Comment by mcoughlin** [comment]
@guynir42 Yeah thats what I am worried about. Plus we know the most recent haven't always been the best due to errors from the source material ;)

**Comment by guynir42** [comment]
@mcoughlin so is there a way to query against older properties? I wonder if this is something that is going to be used?

**Comment by mcoughlin** [comment]
@guynir42 For the moment, we are just building out the capabilities. I think we first will need to make a way to display them and so on, lots of work here to do.

**Comment by guynir42** [comment]
@mcoughlin ok then.

**Pull request merged**